### ISSUE TITLE
Provide feedback in the UI (perhaps in a log) of actions

### ISSUE THREAD

**Issue by stevegrossi** [stevegrossi/sengoku#17]
It can sometimes be hard to follow what the AI (or other players) are doing. Certainly, visual feedback like animations may help here, but I'd also like to add a log of each action so you can inspect what happened if you like.

Options:

1. I could store this in the `GameServer`'s `state`, though that gets copied a lot so I'll want to make sure it doesn't lead to memory issues copying a growing amount of log data.
2. I could store it in a `game_logs` ETS table which would avoid cluttering up the game state, at the cost of a little extra complexity (this feels like the ideal OTP way of doing things)
3. Or, if persistence isn't a requirement, I could simply return a `last_log_item` key in the `state` that gets overwritten after each new action, and which the client-side code appends to a list. The downside here is that if you refresh, you'll lose historical log data, but that may be fine if the goal is only to answer the question "What the hell *just* happened?"

I'm leaning toward 3 as an MVP and can consider 1 or 2 if persistence becomes a requirement.

**Comment by yakryder** [comment]
For 3) you could maybe pop it into local storage if you want benefits of both persistence and off-loading onto the client

**Comment by stevegrossi** [comment]
> I could simply return a last_log_item key in the state

Another idea would be a separate `games:GAME_ID:log` channel, the benefit being that I could push to that log at any point, instead of only as part of state changes.

**Comment by yakryder** [comment]
So one channel for game actions and one channel for game state?

**Comment by stevegrossi** [comment]
Yeah, that's what I'm thinking. We could always send them over the same channel too and differentiate them by type (e.g. `%Message{type: :state_update}`, `%Message{type: :log_update}`) but channels are just processes which are cheap so that seems like premature optimization.

**Comment by stevegrossi** [comment]
I'd also like to incorporate chat somehow (#23) so players can ~~taunt~~ talk to each other (maybe form alliances?) which would be another channel, e.g. `games:GAME_ID:chat` or something.

**Comment by yakryder** [comment]
_Yeah, that's what I'm thinking. We could always send them over the same channel too and differentiate them by type (e.g. %Message{type: :state_update}, %Message{type: :log_update}) but channels are just processes which are cheap so that seems like premature optimization._

I agree, just wanted to make sure I understood your proposed implementation and to hear your thoughts on it.
